deep_qa/models/sequence_tagging/verb_semantics_model.py

```diff
@@ -0,0 +1,96 @@
+from keras.layers import Dense, Input, TimeDistributed
+
+from deep_qa.layers.attention import WeightedSum
+from deep_qa.layers.encoders.bag_of_words import BOWEncoder
+from overrides import overrides
+
+from deep_qa.common.params import Params
+from deep_qa.data.instances.sequence_tagging import concrete_instances
+from deep_qa.training.text_trainer import TextTrainer
+from deep_qa.training.models import DeepQaModel
+
+
+class VerbSemanticsModel(TextTrainer):
+    """
+    This ``VerbSemanticsModel`` takes as input a sentence and query.
+    Query includes verb and entity spans in the sentence.
+    The model predicts (state-change-type, argument-tags)
+    where, state-change-type: denotes what is happening to the entity, e.g. CREATE/DESTROY/MOVE
+    argument-tags: denotes arguments of the state change, e.g. source and destination location of the entity.
+    
+    We use BOW encoder along with a dense layer to predict state-change-type.
+    We use stacked seq2seq encoders followed by a dense layer to predict argument-tags. 
+    
+    Parameters
+    ----------
+    num_stacked_rnns : int, optional (default: ``1``)
+        The number of ``seq2seq_encoders`` that we should stack on top of each other before
+        predicting tags.
+    instance_type : str
+        Specifies the instance type, currently the only supported type is "VerbSemanticsInstance",
+        which defines things like how the input data is formatted and tokenized.
+    """
+    def __init__(self, params: Params):
+        self.num_stacked_rnns = params.pop('num_stacked_rnns', 1)
+        instance_type = params.pop('instance_type', "VerbSemanticsInstance")
+        self.instance_type = concrete_instances[instance_type]
+        super(VerbSemanticsModel, self).__init__(params)
+
+    @overrides
+    def _instance_type(self):  # pylint: disable=no-self-use
+        return self.instance_type
+
+    @overrides
+    def _build_model(self):
+        # Input: (sentence, verb-span, entity-span)
+        # Output: (state-change-type, argument-tags)
+
+        # shape: (batch_size, text_length)
+        sentence_input = Input(shape=self._get_sentence_shape(), dtype='int32', name='word_array_input')
+        verb_input = Input(shape=self._get_sentence_shape(), dtype='int32', name='verb_array_input')
+        entity_input = Input(shape=self._get_sentence_shape(), dtype='int32', name='entity_array_input')
+
+        # shape: (batch_size, text_length, embedding_dim)
+        sentence_embedding = self._embed_input(sentence_input)
+        verb_embedding = self._embed_input(verb_input)
```

Comment:
Remember here that your verb and entity embeddings are actually indices of the sentence positions where the verb/entity appear in the sentence - so at the moment, you are embedding either a 0 or a 1, which I think is probably not your intention. We can go through this tomorrow or something.